src/interfaces.d.ts

```diff
@@ -0,0 +1,59 @@
+import { EventObject } from 'dojo-core/interfaces';
+import Promise from 'dojo-core/Promise';
+import DefaultRoute from './DefaultRoute';
+import PathRule from './PathRule';
+
+export interface CancelableNavigationArgs extends NavigationArgs {
+	preventDefault(): void;
+}
+
+export interface DefaultRouteArgs extends RouteHandlers {
+	enter(path: string): void | Promise<void>;
+}
+
+// TODO: This interface exists mostly to allow classes like PathRegistry to
+// consume both RouteGroup and Route classes without the liberal use of `<any>`.
+// Is there a cleaner way to implement this?
+export interface MatchableRoute extends RouteHandlers {
+	match?: (path: string) => NavigationArgs;
+	path?: string;
+	removeRoute?: (route: MatchableRoute | string) => MatchableRoute;
+	rule?: PathRule;
+}
+
+export interface NavigationArgs extends EventObject {
```

Comment:
@devpaul originally commented (before I renamed this file):


I feel we should rename this type. Arguments are typically only passed into methods; however, NagivationArgs are returned from PathRegistry#match() and altered in Router#go().

> Maybe NavigationPath, although that might be a bit ambiguous it is being held by the PathRegistry.
> Another option could be NavigationDescriptor."